config/src/vespa/config/frt/frtsource.cpp

```diff
@@ -79,33 +81,48 @@ FRTSource::RequestDone(FRT_RPCRequest * request)
         LOG(debug, "request aborted, stopping");
         return;
     }
-    assert(_currentRequest);
+    std::shared_ptr<FRTConfigRequest> configRequest;
+    {
+        std::lock_guard guard(_lock);
+        auto found = _inflight.find(request);
+        assert(found != _inflight.end());
+        configRequest = found->second;
+    }
     // If this was error from FRT side and nothing to do with config, notify
     // connection about the error.
     if (request->IsError()) {
-        _currentRequest->setError(request->GetErrorCode());
+        configRequest->setError(request->GetErrorCode());
+    }
+    _agent->handleResponse(*configRequest, configRequest->createResponse(request));
+    {
+        std::lock_guard guard(_lock);
+        _inflight.erase(request);
+        _cond.notify_all();
     }
-    _agent->handleResponse(*_currentRequest, _currentRequest->createResponse(request));
     LOG(spam, "Calling schedule");
     scheduleNextGetConfig();
 }
 
 void
 FRTSource::close()
 {
+    RequestMap inflight;
     {
         std::lock_guard guard(_lock);
         if (_closed)
             return;
         LOG(spam, "Killing task");
         _task->Kill();
+        inflight = _inflight;
     }
     LOG(spam, "Aborting");
-    if (_currentRequest.get() != NULL)
-        _currentRequest->abort();
-    LOG(spam, "Syncing");
-    _connectionFactory->syncTransport();
```

Comment:
you probably want to keep this one to ensure the transport thread has exited this object after close is called (after waiting for inflight requests)